### PR TITLE
Feature better dispatch

### PR DESCRIPTION
**Pull request by Wikunia**
First implementation of #60 
Use the MOI way of dispatching for all constraints.

### CODE REVIEW

**Comment by Wikunia** [comment]
Interested in a review: @matbesancon :wink:

File: src/ConstraintSolver.jl [hunk 1]
```diff
@@ -56,29 +56,33 @@ struct NoObjective <: ObjectiveFunction end
...
     hash                :: UInt64
-    SingleVariableConstraint() = new()
 end
 
 struct AllDifferentSet <: MOI.AbstractVectorSet
     dimension :: Int64
 end
 
+struct EqualSet <: MOI.AbstractVectorSet
+    dimension :: Int64
```
Comment (matbesancon) [inline src/ConstraintSolver.jl]:
```suggestion
    dimension :: Int
```

File: src/ConstraintSolver.jl [hunk 2]
```diff
@@ -56,29 +56,33 @@ struct NoObjective <: ObjectiveFunction end
...
     rhs                 :: Int
     hash                :: UInt64
-    SingleVariableConstraint() = new()
 end
 
 struct AllDifferentSet <: MOI.AbstractVectorSet
     dimension :: Int64
 end
 
+struct EqualSet <: MOI.AbstractVectorSet
```
Comment (matbesancon) [inline src/ConstraintSolver.jl]: EqualSet is just taking the dimensions but NotEqual takes a value?

Comment (Wikunia) [reply src/ConstraintSolver.jl]: `EqualSet` and `AllDifferentSet` are used in this way:
`vars in Set` but `NotEqualSet` is used for `!=` which means it is a scalar.

File: src/util.jl [hunk 3]
```diff
@@ -33,3 +33,23 @@ function get_safe_lower_threshold(com::CS.CoM, val, divider)
...
+"""
+    linear_combination_to_saf(lc::LinearCombination)
+
+Convert a LinearCombination to a ScalarAffineFunction and return the SAF + the used type
+"""
+function linear_combination_to_saf(lc::LinearCombination)
+    T = eltype(lc.coeffs)
+    sat = [MOI.ScalarAffineTerm{T}(lc.coeffs[i],MOI.VariableIndex(lc.indices[i])) for i=1:length(lc.indices)]
+    return SAF{T}(sat, zero(T)), T
+end
```
Comment (matbesancon) [inline src/util.jl]:
```suggestion
end

```

**Pull request merged**